src/screen/RequestScreen.js

```diff
@@ -195,50 +192,72 @@ class RequestScreen extends Screen {
 	load(path) {
 		const cache = this.getCache();
 		if (isDefAndNotNull(cache)) {
-			return CancellablePromise.resolve(cache);
+			return Promise.resolve(cache);
 		}
 		let body = null;
 		let httpMethod = this.httpMethod;
-		const headers = new MultiMap();
-		Object.keys(this.httpHeaders).forEach(header => headers.add(header, this.httpHeaders[header]));
+
+		const headers = new Headers();
+
+		Object.keys(this.httpHeaders).forEach(header => {
+			headers.append(header, this.httpHeaders[header]);
+		});
+
 		if (globals.capturedFormElement) {
 			this.addSafariXHRPolyfill();
 			body = new FormData(globals.capturedFormElement);
 			this.maybeAppendSubmitButtonValue_(body);
 			httpMethod = RequestScreen.POST;
 			if (UA.isIeOrEdge) {
-				headers.add('If-None-Match', '"0"');
+				headers.append('If-None-Match', '"0"');
 			}
 		}
+
 		const requestPath = this.formatLoadPath(path);
-		return Ajax
-			.request(requestPath, httpMethod, body, headers, null, this.timeout)
-			.then(xhr => {
-				this.removeSafariXHRPolyfill();
-				this.setRequest(xhr);
-				this.assertValidResponseStatusCode(xhr.status);
-				if (httpMethod === RequestScreen.GET && this.isCacheable()) {
-					this.addCache(xhr.responseText);
-				}
-				xhr.requestPath = requestPath;
-				return xhr.responseText;
+
+		const request = new Request(requestPath, {
+			body: body,
+			credentials: 'include',
+			headers: headers,
+			method: httpMethod
+		});
+
+		this.setRequest(request);
+
+		return Promise.race([
+			fetch(request)
+				.then(resp => {
+					this.removeSafariXHRPolyfill();
+					this.assertValidResponseStatusCode(resp.status);
+					return resp.text();
+				})
+				.then(text => {
+					if (httpMethod === RequestScreen.GET && this.isCacheable()) {
+						this.addCache(text);
+					}
+					return text;
+				}).catch((err) => {
+					throw err;
+				}),
```

Comment:
Does this `catch` have any effect? I would expect the code to behave the same if you delete it, unless I'm mistaken.

Reply:
Yes that seems to be the case - not needed.